app/services/twitter_service.rb

```diff
@@ -19,4 +19,79 @@ def client
   def uids
     client.friend_ids.to_a.map(&:to_s)
   end
+
+  def share!(checkin)
+    if checkin.twitter_url_hash.blank?
+      checkin.update_column(:twitter_url_hash, checkin.generate_url_hash)
+    end
+
+    client.update(tweet_body(checkin))
+  end
+
+  private
+
+  def tweet_body(checkin)
+    data = {
+      work_title: checkin.work.title,
+      episode_number: get_episode_number(checkin),
+      episode_title: get_episode_title(checkin),
+      share_url: "#{ENV['ANNICT_HOST_SHORT']}/r/tw/#{checkin.twitter_url_hash}",
+      share_hashtag: get_share_hashtag(checkin),
+      comment: checkin.comment
+    }
+
+    body = generate_tweet_body(data)
+    data = adjust_tweet_body(body, data)
+    generate_tweet_body(data)
+  end
+
+  def get_episode_number(checkin)
+    checkin.episode.single? ? "" : checkin.episode.number
+  end
+
+  def get_episode_title(checkin)
+    if checkin.episode.single?
+      ""
+    else
+      checkin.episode.title.presence || ""
+    end
+  end
+
+  def get_share_hashtag(checkin)
+    checkin.work.twitter_hashtag.presence || ""
+  end
+
+  def generate_tweet_body(data)
```

Comment:
Assignment Branch Condition size for generate_tweet_body is too high. [16.16/15]